crates/env/src/engine/experimental_off_chain/impls.rs

```diff
@@ -460,4 +460,20 @@ impl TypedEnvBackend for EnvInstance {
         self.engine.random(subject, &mut &mut output[..]);
         scale::Decode::decode(&mut &output[..]).map_err(Into::into)
     }
+
+    fn is_contract<T>(&mut self, _account: &T::AccountId) -> bool
+    where
+        T: Environment,
+    {
+        // always false, as off-chain environment does not support contract instantiation
+        false
+    }
+
+    fn caller_is_origin<T>(&mut self) -> bool
+    where
+        T: Environment,
+    {
+        // always true, as off-chain environment does not support contract instantiation
+        true
+    }
```

Comment:
Not yet AFAIK: #788  
But since it's planned, probably it's better to put `unimplemented!()` here?

Reply:
Depends whether #788 is a blocker to switch to the experimental engine by default (and removing the old one). If yes it is safe to put `unimplemented!` there. If not it would break the examples you added. What do you think @cmichi ?

Reply:
Let's put `unimplemented!()` there for now. I'm currently preparing a PR which will resolve some blockers so that we can switch to the experimental engine, but https://github.com/paritytech/ink/issues/788 will not be in there.